src/compiler/checker.ts

```diff
@@ -7359,6 +7359,11 @@ namespace ts {
                 type = <UnionType>createType(TypeFlags.Union | propagatedFlags);
                 unionTypes.set(id, type);
                 type.types = types;
+                /*
+                Note: This is the alias symbol (or lack thereof) that we see when we first encounter this union type.
+                If there exist both `type T = number | boolean` and `type U = number | boolean`, it is arbitrary which one gets the aliasSymbol.

```

Comment:
I would write this line as
> For aliases of identical unions, eg `type T = A | B; type U = A | B`, the symbol of the first alias encountered is the `aliasSymbol`.

This is technically true even for the language server, but "first alias encountered" depends on the order of actions by the user.